spec/unit/config_files_inspector_spec.rb

```diff
@@ -96,231 +119,335 @@
-        "--configfiles", "--queryformat",
-        "%{NAME}-%{VERSION}\n",
-        :stdout => :capture
-      ).and_return(output)
+                            "stat", "--printf", "%a:%U:%G:%u:%g:%n\\n", *files,
+                            :stdout => :capture
+                        ).and_return(stats_output)
     end
 
-    def expect_data_gather_cmds(system,files,stats_output)
-      expect(system).to receive(:run_command).with(
-        "stat", "--printf", "%a:%U:%G:%u:%g:%n\\n", *files,
-        :stdout => :capture
-      ).and_return(stats_output)
+
+    let(:expected_package_list) { ["apache2-2.4.6", "open-iscsi-2.0.873"] }
+    subject { ConfigFilesInspector.new(system, description) }
+
+    describe "check requirements" do
+      it "returns true if requirements are fulfilled" do
+        expect(system).to receive(:check_requirement).with("rpm", "--version")
+        expect(system).to receive(:check_requirement).with("stat", "--version")
+
+        requirements_fulfilled = subject.check_requirements(false)
+        expect(requirements_fulfilled).equal?(true)
+      end
+
+      it "returns true if requirements are fulfilled for extraction" do
+        expect(system).to receive(:check_requirement).with("rpm", "--version")
+        expect(system).to receive(:check_requirement).with("stat", "--version")
+        expect(system).to receive(:check_requirement).with("rsync", "--version")
+
+        requirements_fulfilled = subject.check_requirements(true)
+        expect(requirements_fulfilled).equal?(true)
+      end
     end
 
-    def expect_inspect_configfiles(system,extract)
-      expect_requirements(system)
-      if(extract)
-        expect(system).to receive(:check_requirement).with(
-          "rsync", "--version"
-        )
+
+    describe "#packages_with_config_files" do
+      it "returns a list of packages with config files" do
+        expect(system).to receive(:run_command).with("rpm", "-qa", "--configfiles", "--queryformat",
+                                                     "%{NAME}-%{VERSION}\n", {:stdout => :capture}).and_return(rpm_qa_output_test1)
+
+        package_list = subject.packages_with_config_files
+
+        expect(package_list).to match_array expected_package_list
       end
-      expect_rpm_qa(system,rpm_qa_output_test1)
-      rpm_test_data = [
-        ["apache2-2.4.6", rpm_v_apache_output],
-        ["open-iscsi-2.0.873", rpm_v_openiscsi_output]
-      ]
-      rpm_test_data.each do |pkg_name, output|
-        expect(system).to receive(:run_command).with(
-          *base_cmdline, pkg_name,
-          :stdout => :capture
-        ).and_raise(Cheetah::ExecutionFailed.new(nil, nil, output, nil))
+
+      it "returns a unique list of packages with config files" do
+        expect(system).to receive(:run_command).with("rpm", "-qa", "--configfiles", "--queryformat",
+                                                     "%{NAME}-%{VERSION}\n", {:stdout => :capture}).and_return(rpm_qa_output_test3)
```

Comment:
Line is too long. [131/100]<br>Redundant curly braces around a hash parameter.<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.